src/lazy/utils/index.ts

```diff
@@ -0,0 +1,44 @@
+import { Loader } from '../loader';
+
+declare const __webpack_require__: any;
+declare function __webpack_get_script_filename__(chunkId: string): string;
+declare const __webpack_installed_chunks__: { [k: string]: number };
+
+function requirePreload(chunkId: string) {
+  const href = __webpack_get_script_filename__(chunkId);
+
+  // If chunk already installed or preloaded
+  if (
+    __webpack_installed_chunks__[chunkId] === 0 ||
+    document.head?.querySelector(`link[href="${href}"]`)
+  )
+    return;
+
+  const link = document.createElement('link');
+  link.rel = 'preload';
+  link.as = 'script';
+  link.href = href;
+  document.head?.appendChild(link);
+}
+
+export function preloadAsset(loader: Loader<unknown>) {
+  if (
+    typeof __webpack_require__ === 'undefined' ||
```

Comment:
Have you considered how this should work with parcel? Most libraries fallback to just using the `loader` directly

Reply:
Added the `loader` call as last "strategy"